runtime/src/metrics_server.c

```diff
@@ -81,6 +66,21 @@ metrics_server_handler(void *arg)
 	/* Intermediate cast to integral value of 64-bit width to silence compiler nits */
 	int client_socket = (int)(long)arg;
 
+	/* Duplicate fd so fclose doesn't close the actual client_socket */
```

Comment:
I discovered a bug when testing. I was trying to route before spawning a thread, but fgets can block. This could take down the listener core. Moving this into the thread resolved the issue.

Reply:
makes sense